### ISSUE TITLE
[Octave] template_matrix.i (and STL templates) not working

### ISSUE THREAD

**Issue by ibell** [swig/swig#354]
I am trying to run simple examples like that shown below in swig for octave, I get the same failure message both in windows and ubuntu.  So its not a windows-related bug.  It would be fantastic to get STL wrappers working in octave so you can do something like
```
func([1,2,3]) 
```
rather than having to create and populate the intermediate ``vectori`` object in a multistep process.  This works great in python, c#, etc....

I noticed too that this test (template_matrix.i) is getting skipped on travis: https://travis-ci.org/swig/swig/jobs/54018779

```
C:\swigwin-3.0.5>swig -version

SWIG Version 3.0.5

Compiled with i586-mingw32msvc-g++ [i586-pc-mingw32msvc]

Configured options: +pcre

Please see http://www.swig.org for reporting bugs and further information

C:\swigwin-3.0.5>swig -octave Examples\test-suite\template_matrix.i
C:\swigwin-3.0.5\Lib\octave\std_common.i(71) : Error: Syntax error in input(1).

C:\swigwin-3.0.5>
```

**Comment by ojwb** [comment]
Have you tried `git bisect` to try to work out when this stopped working/if this has even worked?

**Comment by ibell** [comment]
That's the thing, I am not sure if it ever did work, but it sure would be
nice if it did.  I can try to bisect a bit back and see, but I have no idea
how far back I have to go

**Comment by wsfulton** [comment]
Getting this to work would require someone with Octave C api knowledge to contribute it this support.

Regarding template_matrix.i, this is a python only test... it is only in Examples/test-suite/python/Makefile.in and not in the Examples/test-suite/common.mk file. Probably it could be moved over to be a common test, as it doesn't contain any Python specifics, but would need testing first.

**Comment by wsfulton** [comment]
Isn't the title misleading as STL templates do work, there are numerous runtime tests testing the STL? See the Examples/test-suite/octave/*runme.m files.

**Comment by ibell** [comment]
Yes, I guess I meant native support, not requiring first the construction
of an intermediate class and populating the class data using push_back() or
accessors.

You are right that STL containers are supported, just not with the nice
interface I have become accustomed to from python.

**Comment by ojwb** [comment]
Still the case in git master.

`Examples/test-suite/octave/li_std_containers_overload_runme.m` includes a commented out testcase (and isn't run by default) but enabling it and commenting out it doesn't actually work (as the comment in the testcase says).

(The non-commented out testcases in that file pass though, so I'll push a change to enable it.)

**Comment by ojwb** [comment]
> I'll push a change to enable it

I wasn't sure if I'd actually done that and couldn't spot an obvious change enabling it, but running `make check-octave-test-suite` does now run it:

```
checking octave testcase li_std_containers_overload (with run test)
```

We still don't support native wrapping AFAICS though, but I'd agree with wsfulton that really needs someone with Octave C API knowledge to contribute a patch.
